lib/matplotlib/ticker.py

```diff
@@ -475,11 +475,16 @@ def __init__(self, useOffset=None, useMathText=None, useLocale=None):
         self._usetex = mpl.rcParams['text.usetex']
         if useMathText is None:
             useMathText = mpl.rcParams['axes.formatter.use_mathtext']
-            if useMathText is False and "cmr10" in mpl.rcParams['font.family']:
-                _api.warn_external(
-                    'cmr10 font should ideally be used with '
-                    'mathtext, set axes.formatter.use_mathtext to True'
-                )
+            if useMathText is False:
+                ff = mpl.rcParams["font.family"]
+                fm = ("serif", "sans-serif", "cursive", "monospace", "fantasy")
+                if "cmr10" in ff or any(
+                    "cmr10" in mpl.rcParams[f"font.{family}"] for family in fm
+                ):
```

Comment:
I think this warning is overly broad: if e.g. `font.family = "sans-serif" and `font.serif = "cmr10"`, you don't want to emit a warning (because we're not going to use the serif font anyways).  What you really want to check is probably something more like
```python
mpl.font_manager.findfont(mpl.font_manager.FontProperties(mpl.rcParams["font.family"])) \
    == <path to cmr10.ttf>
```

Reply:
That makes so much more sense!
I'll update.